src/agent/communicator/src/communicator.cpp

```diff
@@ -6,16 +6,21 @@
 #include <iostream>
 #include <nlohmann/json.hpp>
 
-namespace net = boost::asio;
 using tcp = boost::asio::ip::tcp;
 using json = nlohmann::json;
 
 namespace communicator
 {
 
-    Communicator::Communicator()
+    Communicator::Communicator(const std::function<std::string(std::string, std::string)> GetStringConfigValue)
+    : m_managerIp("")
```

Comment:
It's not necessary to set them to `""` that's the default value of `std::string`, so when the constructor's function body starts, they will already have been default constructed.

Reply:
Done